src/lib.rs

```diff
@@ -108,13 +113,15 @@ mod sink;
 mod spatial_sink;
 
 pub mod buffer;
+#[cfg(any(feature = "wav", feature = "flac", feature = "vorbis"))]
 pub mod decoder;
 pub mod dynamic_mixer;
 pub mod queue;
 pub mod source;
 
 /// Plays a sound once. Returns a `Sink` that can be used to control the sound.
 #[inline]
+#[cfg(any(feature = "wav", feature = "flac", feature = "vorbis"))]
```

Comment:
It's not great to remove this function if you don't enable any decoder. I think the `DecoderError` should just be exposed in any situation instead.